jfix-stdlib-concurrency/src/main/java/ru/fix/stdlib/concurrency/threads/ReschedulableScheduler.java

```diff
@@ -190,11 +220,15 @@ private synchronized void checkPreviousScheduleAndRestartTask(Schedule schedule)
         }
 
         synchronized ScheduledFuture<?> launch() {
+            this.scheduleListener = (oldVal, newVal) -> this.checkPreviousScheduleAndRestartTask(newVal);
+            this.schedule.addListener(this.scheduleListener);
             Schedule schedule = this.schedule.get();
 
             this.scheduledFuture = schedule(this, schedule, this.startDelay);
             previousSchedule = schedule;
 
+            log.trace("scheduledFuture={} with hash={} is launched",
+                    scheduledFuture, System.identityHashCode(scheduledFuture));
```

Comment:
You added trace to launch(), but did not add trace to checkPreviousScheduleAndRestartTask method that actually cancel and reschedules new task